app/src/main/java/com/fmsys/snapdrop/MainActivity.java

```diff
@@ -190,10 +190,9 @@ protected void onCreate(final Bundle savedInstanceState) {
                 final Iterator<JavaScriptInterface.FileHeader> iterator = downloadFilesList.iterator();
                 while (iterator.hasNext()) {
                     final JavaScriptInterface.FileHeader file = iterator.next();
+                    copyTempToDownloads(file);
                     if (file.getSize().equals(String.valueOf(contentLength))) {
```

Comment:
Let me explain the background of this method. When using classic snapdrop instances, all files are downloaded without confirmation. By checking the file size, I made sure that only files approved by the user are actually copied into the user storage. With this change, all already downloaded files will be stored as soon as the user approves one of these files. This might be an unwanted behavior. 

OTOH @anggrayudi has started refactoring this workflow anyway in #202, unfortunately I never found time to review it (sorry for that 🙏). #202 and #271 would both make sense especially when moving towards pairdrop

Reply:
I did not like unintended files being sent and downloaded automatically as well so there is a check at PairDrop too: https://github.com/schlagmichdoch/PairDrop/blob/master/public/scripts/network.js#L440

If you want to have another check before downloading we could maybe iterate through all files in the `downloadFilesList` and add up the result of `file.getSize()` and compare that sum with `contentLength`. This would require iterating through the list twice but I guess that would be ok as the list is not long

Reply:
Also, PairDrop has implemented the suggestions @anggrayudi made: 
- confirmation before sending files
- image previews are sent as thumbnails created by the client and not by the server

Reply:
I've applied this PR to #202. As soon as I have the ok from @anggrayudi I'll merge that PR which should achieve pairdrop compatibility and provides better download speed.

Reply:
Awesome! Have you already tested it with https://pairdrop-dev.onrender.com/ ? I will wait for that PR to be merged and then add the compatibility changes to PairDrop including a mention in the docs.

Reply:
Single-file and multi-file transfer seems to work fine. However, the "redownload" dialog is confusing IMHO. Would it be possible to hide that dialog?

(maybe only for Snapdrop for Android, I could add some unique identifier to the user argent)

Reply:
The thing is that files are normally downloaded automatically after a succesful transfer. If the transfer takes very long it is not possible to download automatically as the user interaction may be inactive so the "Download Again" button is a fallback.

I want the user to be notified that the file is successfully transferred and show a preview for images, videos and audio.

With the app you have the control over the download process itself and you can ensure the download is successfully. So I guess it would be easier if this is done in this app. Instead of hiding the whole dialog I would probably hide the "Download again" button:
<img width="200px" src="https://user-images.githubusercontent.com/52242352/224847681-49ea2162-c666-43a7-bbad-697f72cbf6fa.png"/>

Probably adding the following to the `init.js` would be enough
``` javascript
document.getElementById('download-btn').setAttribute("display", "none");
```

Reply:
there aleady is a snack-/toastbar that is indicating a successful transfer and has the possibility to open the file so i would just completly get rid of the "received" dialog in the app, no? @fm-sys